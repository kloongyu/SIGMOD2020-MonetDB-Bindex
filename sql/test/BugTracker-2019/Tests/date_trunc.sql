select timestamp '2119-02-17 02:08:12.345678';

select date_trunc('microseconds', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('milliseconds', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('second', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('minute', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('hour', timestamp '2119-02-17 02:08:12.345678');

select date_trunc('day', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('week', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('month', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('quarter', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('year', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('decade', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('century', timestamp '2119-02-17 02:08:12.345678');
select date_trunc('millennium', timestamp '2119-02-17 02:08:12.345678');
