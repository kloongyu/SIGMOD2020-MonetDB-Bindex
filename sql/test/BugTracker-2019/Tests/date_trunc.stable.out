stdout of test 'date_trunc` in directory 'sql/test/BugTracker-2019` itself:


# 08:07:24 >  
# 08:07:24 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33395" "--set" "mapi_usock=/var/tmp/mtest-23934/.s.monetdb.33395" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/branches/default//Linux/var/MonetDB/mTests_sql_test_BugTracker-2019" "--set" "embedded_c=true"
# 08:07:24 >  

# MonetDB 5 server v11.31.14
# This is an unreleased version
# Serving database 'mTests_sql_test_BugTracker-2019', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 31.309 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2019 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://toke.da.cwi.nl:33395/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-23934/.s.monetdb.33395
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.

# 08:07:24 >  
# 08:07:24 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-23934" "--port=33395"
# 08:07:24 >  

#select date_trunc('microseconds', timestamp '2019-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-02-17 02:08:12.346000	]
#select date_trunc('microseconds', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-02-17 02:08:12.346000	]
#select date_trunc('milliseconds', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-02-17 02:08:12.346000	]
#select date_trunc('second', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-02-17 02:08:12.000000	]
#select date_trunc('minute', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-02-17 02:08:00.000000	]
#select date_trunc('hour', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-02-17 02:00:00.000000	]
#select date_trunc('day', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-02-17 00:00:00.000000	]
#select date_trunc('week', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-02-11 00:00:00.000000	]
#select date_trunc('month', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-02-01 00:00:00.000000	]
#select date_trunc('quarter', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-01-01 00:00:00.000000	]
#select date_trunc('year', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2119-01-01 00:00:00.000000	]
#select date_trunc('decade', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2110-01-01 00:00:00.000000	]
#select date_trunc('century', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2101-01-01 00:00:00.000000	]
#select date_trunc('millennium', timestamp '2119-02-17 02:08:12.345678');
% .L2 # table_name
% L2 # name
% timestamp # type
% 26 # length
[ 2001-01-01 00:00:00.000000	]

# 08:07:24 >  
# 08:07:24 >  "Done."
# 08:07:24 >  

